### PR TITLE
attach displayName to forwardRef function as described in React docs

### PR DESCRIPTION
**Pull request by MrLoh**
This addresses #2507

### CODE REVIEW

**Comment by MrLoh** [comment]
ok, I guess it wasn't that easy ;)

**Comment by MrLoh** [comment]
This needs some more testing I think.

File: packages/styled-components/src/models/StyledComponent.js [hunk 1]
```diff
@@ -277,15 +277,18 @@ export default function createStyledComponent(target: Target, options: Object, r
...
    */
-  const WrappedStyledComponent = React.forwardRef((props, ref) => (
+  let WrappedStyledComponent;
+  const forwardRef = (props, ref) => (
     <ParentComponent {...props} forwardedComponent={WrappedStyledComponent} forwardedRef={ref} />
-  ));
+  );
+  forwardRef.displayName = displayName;
+  WrappedStyledComponent = React.forwardRef(forwardRef);
+  WrappedStyledComponent.displayName = displayName;
```
Comment (probablyup) [inline packages/styled-components/src/models/StyledComponent.js]: I guess this one isn't needed if you put it on the underlying one?

Comment (MrLoh) [reply packages/styled-components/src/models/StyledComponent.js]: Yeah but all tests fail otherwise, will need to understand how the tests check this

**Comment by chasemccoy** [comment]
@probablyup Any chance of getting this published? I think it might fix an issue I am running into.

**Comment by probablyup** [comment]
I’ve been in the hospital so I really can’t. But if one of the other core
collaborators wants to they can.

**Comment by MrLoh** [comment]
So sorry to hear that. All the best to you, focus on getting well and thank you so much for your contributions here.

**Comment by chasemccoy** [comment]
@probablyup Oh no! Hope everything is well, and thank you for all your hard work on styled-components!

**Pull request merged**